src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -577,6 +578,12 @@ Copyright (c) .NET Foundation. All rights reserved.
       <Output TaskParameter="AbsolutePaths" PropertyName="RestoreOutputAbsolutePath" />
     </ConvertToAbsolutePath>
 
+    <WarnForMismatchedBaseIntermediateOutputPathTask 
```

Comment:
I'm not convinced that this makes sense.  I could set just `MSBuildProjectExtensionsPath` and leave `BaseIntermediateOutputPath` alone if I wanted.  With this warning, I would feel like I need to set them both always to the same location.  Why even have a different property?

If NuGet only reads `MSBuildProjectExtensionsPath`,  why would it care what the value of `BaseIntermediateOutputPath` is?

Reply:
I agree with Jeff on this, that's why I added it in the follow-up task, becuase I'm not convinced we should add it.

Reply:
@jeffkl @nkolev92 My reasoning is that the common case will be that `BaseIntermediateOutputPath` is set in the body of the project, and the result is not what was expected (ie the expectation was that nothing would be written to `obj` in the project folder.  So the warning helps guide people to the right way of doing what they want to do.

I forgot to allow the warning message to be disabled by setting a property, which is what I was thinking you would do if you actually wanted the paths to be different.

Reply:
I think we should rely on the error in MSBuild we're adding: https://github.com/Microsoft/msbuild/pull/3059

Is that not enough?  I'd be okay with this warning if it was only displayed when BaseIntermediateOutputPath is different and it was modified.  But I don't think it should happen anytime they're different after evaluation...